presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -409,6 +409,15 @@ public PlanOptimizers(
                 ImmutableSet.of(
                         new CreatePartialTopN(),
                         new PushTopNThroughUnion())));
+        builder.add(new IterativeOptimizer(
+                ruleStats,
+                statsCalculator,
+                costCalculator,
+                ImmutableSet.<Rule<?>>builder()
+                        .add(new RemoveRedundantIdentityProjections())
+                        .addAll(new TransformSpatialPredicates(metadata).rules())
```

Comment:
Why it was moved?

Reply:
@kokosing To address the TODO which required a dedicated SpatialJoinNode. This type of rule logically belongs before `AddExchanges`, so I moved it.

```
        // TODO Make PredicatePushDown aware of spatial joins, move TransformSpatialPredicateToJoin	
        // before AddExchanges and update AddExchanges to set REPLICATED distribution for the build side.
```